lib/ansible/modules/system/xfs_quota.py

```diff
@@ -0,0 +1,393 @@
+
+    result = dict(
+        changed=False
+    )
+
+    if not os.path.ismount(mountpoint):
+        module.fail_json(msg='%s is not a mountpoint' % mountpoint, **result)
+
+    mp = get_fs_by_mountpoint(mountpoint)
+    if mp is None:
+        module.fail_json(msg='%s is not a mountpoint or not located on an xfs filesystem.' % mountpoint, **result)
+
+    if quota_type == 'user':
+        type_arg = '-u'
+        quota_default = 'root'
+        if name is None:
+            name = quota_default
+
+        if 'uquota' not in mp['mntopts'] \
+                and 'usrquota' not in mp['mntopts'] \
+                and 'quota' not in mp['mntopts'] \
+                and 'uqnoenforce' not in mp['mntopts'] \
+                and 'qnoenforce' not in mp['mntopts']:
+            module.fail_json(
+                msg='%s is not mounted with the uquota/usrquota/quota/uqnoenforce/qnoenforce option.'
+                    % mountpoint, **result
+            )
+        try:
+            pwd.getpwnam(name)
+        except KeyError as e:
+            module.fail_json(msg='User %s doesn\'t exist.' % name, **result)
+
+    if quota_type == 'group':
+        type_arg = '-g'
+        quota_default = 'root'
+        if name is None:
+            name = quota_default
+
+        if 'gquota' not in mp['mntopts'] and 'grpquota' not in mp['mntopts'] and 'gqnoenforce' not in mp['mntopts']:
+            module.fail_json(
+                msg='%s is not mounted with the gquota/grpquota/gqnoenforce option. (current options: %s)'
+                    % (mountpoint, mp['mntopts']), **result
+            )
+        try:
+            grp.getgrnam(name)
+        except KeyError as e:
+            module.fail_json(msg='User %s doesn\'t exist.' % name, **result)
+
+    elif quota_type == 'project':
+        type_arg = '-p'
+        quota_default = '#0'
+        if name is None:
+            name = quota_default
+
+        if 'pquota' not in mp['mntopts'] and 'prjquota' not in mp['mntopts'] and 'pqnoenforce' not in mp['mntopts']:
+            module.fail_json(msg='%s is not mounted with the pquota/prjquota/pqnoenforce option.' % mountpoint, **result)
+
+        if name != quota_default and not os.path.isfile('/etc/projects'):
+            module.fail_json(msg='/etc/projects doesn\'t exist.', **result)
+
+        if name != quota_default and not os.path.isfile('/etc/projid'):
+            module.fail_json(msg='/etc/projid doesn\'t exist.', **result)
+
+        if name != quota_default and name is not None and get_project_id(name) is None:
+            module.fail_json(msg='%s hasn\'t been defined in /etc/projid.' % name, **result)
+
+        prj_set = True
+        if name != quota_default:
+            cmd = 'project %s' % name
+            r = exec_quota(module, cmd, mountpoint)
+            if r['rc'] != 0:
+                result['cmd'] = cmd
+                result['retvar'] = r
+                module.fail_json(msg='Could not get project state.', **result)
```

Comment:
You probably want to return stdout and stderr on error as well. So:
```python
        rc, stdout, stderr = exec_quota(module, cmd, mountpoint)
        if rc != 0:
            module.fail_json(msg='Could not get project state.', cmd=cmd, rc=rc, stdout=stdout, stderr=stderr)
```

Reply:
But this is already the case...
This is maybe due to one of my earlier commits...